Doc/library/math.rst

```diff
@@ -646,8 +646,27 @@ Constants
 
 .. data:: nan
 
-   A floating-point "not a number" (NaN) value.  Equivalent to the output of
-   ``float('nan')``.
+   A floating-point "not a number" (NaN) value. Equivalent to the output of
+   ``float('nan')``. Specifically, due to the requirements of the `IEEE-754 standard
+   <https://en.wikipedia.org/wiki/IEEE_754>`_, ``math.nan`` and ``float('nan')`` are
+   not considered to equal to any other numeric value, including themselves. To avoid errors
+   when checking whether a number is a NaN, use the :func:`isnan` function to test
+   for NaNs instead of ``is`` or ``==``.
+   Example::
+
+      >>> import math
+      >>> math.nan == math.nan
+      False
+      >>> float('nan') == float('nan')
+      False
+      >>> math.nan is math.nan
+      True
+      >>> float('nan') is float('nan')
```

Comment:
Not too happy about this example because I could see us change this behavior in the future to return the same NaN object.

Reply:
I feel the same way, this example is confusing. This phenomenon is simply due to the differences in function implementations, not related to any standard. 

Maybe we should omit the example about `is` and just keep docs to remind users not use `is` to check if a number is a NaN.